src/main/java/org/broadinstitute/hellbender/engine/AlignmentContext.java

```diff
@@ -122,10 +122,16 @@ public AlignmentContext stratify(final ReadOrientation type) {
      * Splits the given AlignmentContext into a StratifiedAlignmentContext per sample, but referencd by sample name instead
      * of sample object.
      *
+     * @param assumedSingleSample assumed single sample in the file or null if there is none (eg file contains more than 1 sample).
+     *                            If this argument is not null then this method returns <code>Collections.singletonMap(assumedSingleSample, this);</code>
      * @return a Map of sample name to StratifiedAlignmentContext
      *
      **/
     public Map<String, AlignmentContext> splitContextBySampleName(final String assumedSingleSample, final SAMFileHeader header) {
+        if (assumedSingleSample != null){
+            return Collections.singletonMap(assumedSingleSample, this);
```

Comment:
i did a quick search. i don;t think so any of them is making any assumptions one way or the other